packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -212,11 +213,23 @@ abstract class CachingAssetBundle extends AssetBundle {
 
 /// An [AssetBundle] that loads resources using platform messages.
 class PlatformAssetBundle extends CachingAssetBundle {
+
+  /// Creates a [PlatformAssetBundle] with the specified [binaryMessenger].
+  ///
+  /// A default [BinaryMessenger] instance will be used if [binaryMessenger] is
+  /// null.
+  PlatformAssetBundle({this.binaryMessenger}) {
+    binaryMessenger ??= ServicesBinding.instance.provideBinaryMessenger();
+  }
+
+  /// The messenger which sends the bytes for this channel, not null.
+  BinaryMessenger binaryMessenger;
```

Comment:
this should be final

Reply:
Updated to use the _binaryMessenger getter.